lib/rule-options.ts

```diff
@@ -1,19 +1,23 @@
 import traverse from 'json-schema-traverse';
-import type { JSONSchema } from '@typescript-eslint/utils';
+import type {
+  JSONSchema4,
+  JSONSchema4Type,
+  JSONSchema4TypeName,
+} from 'json-schema';
```

Comment:
Needed to switch to the types from json-schema, likely because of this PR affecting the typescript-eslint version: 
* https://github.com/typescript-eslint/typescript-eslint/pull/6963